src/test/java/igym/controllers/GymControllerTest.java

```diff
@@ -281,4 +283,51 @@ void testDeleteGymAlreadyDeleted() throws Exception {
                 verifyNoMoreInteractions(gymService);
         }
 
+        @Test
+        @DisplayName("should return active gyms for a given user ID")
+        void testGetGymsByUserIdSuccess() throws Exception {
+                Gym gymA = new Gym("Gym A");
+                Gym gymB = new Gym("Gym B");
+                gymA.setStatus(Status.active);
+                gymB.setStatus(Status.active);
+                List<Gym> gyms = List.of(gymA, gymB);
+
+                when(gymService.findGymsByUserId(userId)).thenReturn(gyms);
+
+                mockMvc.perform(get("/api/users/{userId}/gyms", userId))
+                                .andExpect(status().isOk())
+                                .andExpect(jsonPath("$.length()").value(2))
+                                .andExpect(jsonPath("$[0].name").value("Gym A"))
+                                .andExpect(jsonPath("$[1].name").value("Gym B"));
+
+                verify(gymService).findGymsByUserId(userId);
+        }
+
+        @Test
+        @DisplayName("should return 404 when user is not found")
+        void testGetGymsByUserId_userNotFound() throws Exception {
```

Comment:
```suggestion
        void testGetGymsByUserIdUserNotFound() throws Exception {
```
Let's define a pattern to method names. Or camelCase, or snake_case, but not mixing both patterns